### PR TITLE
fix: strict types error

### LINKED ISSUE #34: Fix remaining strict types errors
With Drupal core moving towards strict types, [more info](https://github.com/localgovdrupal/localgov/issues/685) we should fix the remaining errors suppressed in the phpstan baseline.

```
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  Line   modules/contrib/localgov_menu_link_group/localgov_menu_link_group.module
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  91     Parameter #2 $callback of function array_walk expects callable(Drupal\Core\Entity\EntityInterface, int|string): mixed, array{Drupal\localgov_menu_link_group\MenuLinkGrouper, 'groupChildMenuLinks'} given.
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  Line   modules/contrib/localgov_menu_link_group/src/Controller/LocalGovMenuLinkGroupListBuilder.php
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  51     Parameter #1 $group of method Drupal\localgov_menu_link_group\Controller\LocalGovMenuLinkGroupListBuilder::determineParentMenuLinkLabel() expects Drupal\localgov_menu_link_group\Entity\LocalGovMenuLinkGroupInterface,
         Drupal\Core\Entity\EntityInterface given.
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

 ------ ------------------------------------------------------------------------------------------------------------------------
  Line   modules/contrib/localgov_menu_link_group/src/Form/LocalGovMenuLinkGroupForm.php
 ------ ------------------------------------------------------------------------------------------------------------------------
  80     Dynamic call to static method Drupal\localgov_menu_link_group\Form\LocalGovMenuLinkGroupForm::prepareMenuLinkOption().
 ------ ------------------------------------------------------------------------------------------------------------------------

 ------ -------------------------------------------------------------------------------
  Line   modules/contrib/localgov_menu_link_group/tests/src/Kernel/GroupAccessTest.php
 ------ -------------------------------------------------------------------------------
  64     Class DOMXPath referenced with incorrect case: DomXPath.
  79     Class DOMXPath referenced with incorrect case: DomXPath.
 ------ -------------------------------------------------------------------------------

 ------ -------------------------------------------------------------------------------------
  Line   modules/contrib/localgov_menu_link_group/tests/src/Kernel/GroupConfigImportTest.php
 ------ -------------------------------------------------------------------------------------
  45     Call to function in_array() requires parameter #3 to be set.
  72     Call to function in_array() requires parameter #3 to be set.
 ------ -------------------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by millnut**
## What does this change?

Fixes #34

### CODE REVIEW

File: src/Controller/LocalGovMenuLinkGroupListBuilder.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function buildHeader() {
    * {@inheritdoc}
    */
   public function buildRow(EntityInterface $group) {
-
+    /** @var \Drupal\localgov_menu_link_group\Entity\LocalGovMenuLinkGroup $group */
```
Comment (millnut) [inline src/Controller/LocalGovMenuLinkGroupListBuilder.php]: add this for phpstan to understand the type return buildRow requires EntityInterface but inspecting the types for `$group` LocalGovMenuLinkGroup is returned. This comment helps phpstan and IDEs understand what is then passed further down

File: src/Form/LocalGovMenuLinkGroupForm.php [hunk 2]
```diff
@@ -77,7 +77,7 @@ public function form(array $form, FormStateInterface $form_state) {
       '#value' => $parent_menu_name,
     ];
 
-    $parent_menu_link_option = $this->prepareMenuLinkOption($parent_menu_link, $parent_menu_name);
+    $parent_menu_link_option = self::prepareMenuLinkOption($parent_menu_link, $parent_menu_name);
```
Comment (millnut) [inline src/Form/LocalGovMenuLinkGroupForm.php:80]: Fixes this error

```
Dynamic call to static method Drupal\localgov_menu_link_group\Form\LocalGovMenuLinkGroupForm::prepareMenuLinkOption().
```

File: src/Controller/LocalGovMenuLinkGroupListBuilder.php (hunk 1 above)
Comment (ekes) [reply src/Controller/LocalGovMenuLinkGroupListBuilder.php]: assert() works the same and is nicer though ;-)

Comment (finnlewis) [reply src/Controller/LocalGovMenuLinkGroupListBuilder.php]: Just discussing in Merge Tuesday, and we acknowledge that Drupal Core uses these sort of comments, but we think even core is moving towards using PHP assert(). 

Could we change this please @millnut ?

Comment (millnut) [reply src/Controller/LocalGovMenuLinkGroupListBuilder.php]: 👍 this has now been updated

**Review by ekes — APPROVED** [review]

**Pull request merged**